File: design-patterns/java/java-facade-design-pattern/Facade.java
// Facade class provides a simplified interface to control multiple subsystems (AirConditioning, Door, Light, TV).
public class Facade {
    private AirConditioning airConditioner;
    private Door door;
    private Light light;
    private TV tv;

    public Facade(){
        airConditioner = new AirConditioning();
        door = new Door();
        light = new Light();
        tv = new TV();
    }


    public void turnOnAirCon(){airConditioner.on();}
    public void turnOffAirCon(){airConditioner.off();}
    public void adjustAirCon(int degree){airConditioner.adjust(degree);}

    public void unlockDoor(){door.unlock();}
    public void lockDoor(){door.lock();}

    public void turnOnAllLights(){light.onAll();}
    public void turnOffAllLights(){light.offAll();}

    public void turnOnTv(){tv.on();}
    public void turnOffTv(){tv.off();}

    public void sleepMode(){
        System.out.println("Activating sleep mode...");
        airConditioner.off();
        door.lock();
        light.offAll();
        tv.off();
        System.out.println("Sleep mode activated!");
    }
}

File: design-patterns/python/python-facade-design-pattern/Facade.py
from Light import Light
from Door import Door
from TV import TV
from AirConditioning import AirConditioning

class Facade:
    def __init__(self):
        self.air_conditioner = AirConditioning()
        self.door = Door()
        self.light = Light()
        self.tv = TV()

    def turn_on_air_con(self):
        self.air_conditioner.on()

    def turn_off_air_con(self):
        self.air_conditioner.off()

    def adjust_air_con(self, degree):
        self.air_conditioner.adjust(degree)

    def unlock_door(self):
        self.door.unlock()

    def lock_door(self):
        self.door.lock()

    def turn_on_all_lights(self):
        self.light.on_all()

    def turn_off_all_lights(self):
        self.light.off_all()

    def turn_on_tv(self):
        self.tv.on()

    def turn_off_tv(self):
        self.tv.off()

    def sleep_mode(self):
        print("Activating sleep mode...")
        self.air_conditioner.off()
        self.door.lock()
        self.light.off_all()
        self.tv.off()
        print("Sleep mode activated!")
